07-routing-gossip.md

```diff
@@ -510,16 +524,197 @@ An endpoint node:
+  - if does not maintain up-to-date channel information for `chain_hash`:
+	- MUST set `complete` to 0.
+  - otherwise:
+	- SHOULD set `complete` to 1.
+
+#### Rationale
+
+Future nodes may not have complete information; they certainly won't have
+complete information on unknown `chain_hash` chains.  While this `complete`
+field cannot be trusted, if 0 ir does indicate that the sender should search
+elsewhere for additional data.
+
+The explicit `reply_short_channel_ids_end` message means that the receiver can
+indicate it doesn't know anything, and the sender doesn't need to rely on
+timeouts.  It also causes a natural ratelimiting of queries.
+
+### The `query_channel_range` and `reply_channel_range` Messages
+
+1. type: 263 (`query_channel_range`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`4`:`first_blocknum`]
+    * [`4`:`number_of_blocks`]
+
+1. type: 264 (`reply_channel_range`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`4`:`first_blocknum`]
+    * [`4`:`number_of_blocks`]
+    * [`1`:`complete`]
+    * [`2`:`len`]
+    * [`len`:`encoded_short_ids`]
+
+This allows a query for channels within specific blocks.
+
+#### Requirements
+
+The sender of `query_channel_range`:
+  - MUST NOT send this if it has sent a previous `query_channel_range` to this peer and not received `reply_channel_range`.
+  - MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that it wants the `reply_channel_range` to refer to
+  - MUST set `first_blocknum` to the first block it wants to know channels for
+  - MUST set `number_of_blocks` to 1 or greater.
+
+The receiver of `query_channel_range`:
+  - if it has not sent `reply_channel_range` to a previously received `query_channel_range` from this sender:
+    - MAY fail the connection.
+  - MUST respond with one or more `reply_channel_range` whose combined range
+	cover the requested `first_blocknum` to `first_blocknum` plus
+	`number_of_blocks` minus one.
+  - For each `reply_channel_range`:
+    - MUST set with `chain_hash` equal to that of `query_channel_range`,
+    - MUST encode a `short_channel_id` for every open channel it knows in blocks `first_blocknum` to `first_blocknum` plus `number_of_blocks` minus one.
+    - MUST limit `number_of_blocks` to the maximum number of blocks whose
+      results could fit in `encoded_short_ids`
+    - if does not maintain up-to-date channel information for `chain_hash`:
+      - MUST set `complete` to 0.
+    - otherwise:
+      - SHOULD set `complete` to 1.
+
+#### Rationale
+
+A single response might be too large for a single packet, and also a peer can
+store canned results for (say) 1000-block ranges, and simply offer each reply
+which overlaps the ranges of the request.
+
+### The `gossip_timestamp_range` Message
+
+1. type: 265 (`gossip_timestamp_range`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`4`:`first_timestamp`]
+    * [`4`:`timestamp_range`]
+
```

Comment:
`range` here has not the same meaning as in `query_channel_range`/`reply_channel_range`, `gossip_timestamp_filter` would be more appropriate ?

Reply:
Agreed.

Reply:
Just to clarify, this the `timestamp_range` is in seconds, right? May be good to note that here to be precise.